Initial indieGala support

kohend opened the pull request:
* list
* install
* launch (only with wine, not proton
* freecarnival binary for Linux Login needs to be done manually with freecarnival
no custom settings dir

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

kohend commented:
#1096

flavioislima commented:
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/main/doc/cla.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

kohend commented:
I have read the CLA Document and I hereby sign the CLA

arielj commented:
I'm trying to test this but I can't log in. I get logged in in the store but heroic keeps showing me as not logged in. I don't see any error anywhere that could help, not sure what could be useful for me to try.

kohend commented:
Are you logging in through the stores nenu? Since I haven't gotten the login to work through the nanage accounts flow. Are you logging in with Google or something?

arielj commented:
> Are you logging in through the stores nenu? Since I haven't gotten the login to work through the nanage accounts flow. Are you logging in with Google or something?

right, I understand now the last element listed in the description, I was trying to login using Heroic, but I see it needs to be done using the freecarnival binary in the console

ignore my comment then, it was my using something not yet implemented

kohend commented:
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/309245/ebb280b1-e599-4f83-ae66-9b1094f3992e)
No, I implemented the login since, but perhaps after you pulled the code? added a screenshot to show where to login from.

Gustash commented:
So, a couple of things on this early stage:

- The login flow is counter-intuitive to the other stores. You should handle login in `Manage Accounts` instead. No other stores log you in by using the `Stores` tab.

- You are not creating the `carnival_config` folder, everything breaks if the folder does not exist.

- Login does not work if there isn't an existing `user.yml` file in `carnival_config`. To get that, you should call any FreeCarnival command that syncs the authentication. There isn't a dedicated command right now, but we can add a `refresh/sync` command. Alternatively, you can just call `library`, but it seems wasteful to request so much data.
